python/triton/experimental/gluon/language/amd/cdna3/__init__.py

```diff
@@ -29,24 +29,45 @@ def _verify_buffer_ops(ptr, offsets, mask=None, other=None):
 
 
 def _verify_element_type_in_buffer_atomic(op, elem_type, arch):
-    supported_types = [ttgl.float16, ttgl.float32, ttgl.bfloat16, ttgl.float64, ttgl.int32, ttgl.int64]
+    supported_types = [
+        ttgl.float16, ttgl.float32, ttgl.bfloat16, ttgl.float64, ttgl.int32, ttgl.int64, ttgl.uint32, ttgl.uint64
+    ]
     assert elem_type in supported_types, f"{elem_type} is not supported in buffer atomic on {arch}."
+
     op = _atomic_op_str_to_op[_unwrap_if_constexpr(op)]
-    assert arch != "cdna3" or op != ir.ATOMIC_OP.FADD or elem_type != ttgl.bfloat16, "Buffer atomic fadd with bf16 is not supported on CDNA3"
+    if op in [ir.ATOMIC_OP.AND, ir.ATOMIC_OP.OR, ir.ATOMIC_OP.XOR, ir.ATOMIC_OP.XCHG]:
+        assert elem_type in [ttgl.int32, ttgl.int64], f"{op} with {elem_type} is not supported on CDNA3 and CDNA4"
 
+    if op in [ir.ATOMIC_OP.UMAX, ir.ATOMIC_OP.UMIN, ir.ATOMIC_OP.ADD]:
+        assert elem_type in [ttgl.uint32, ttgl.uint64], f"{op} with {elem_type} is not supported on CDNA3 and CDNA4"
 
-def _buffer_atomic_rmw_impl(op, ptr, offsets, value, arch, mask, sem, scope, _semantic):
-    _verify_element_type_in_buffer_atomic(op, value.type.scalar, arch)
+    if op in [ir.ATOMIC_OP.MAX, ir.ATOMIC_OP.MIN]:
+        assert elem_type in [ttgl.int32, ttgl.int64,
+                             ttgl.float64], "smax only support int32, int64 and fp64 on CDNA3 and CDNA4"
+
+    if op == ir.ATOMIC_OP.FADD:
+        if elem_type is ttgl.bfloat16:
+            assert arch != "cdna3", "Buffer atomic fadd with bf16 is not supported on CDNA3"
```

Comment:
`assert arch == "cdna4", "Buffer atomic fadd with bf16 is only supported on CDNA4"` to be more restrictive. So that in the future if we call this function for some other architecture it won't accidentally allow other cases.